### PR TITLE
Add new manipulation get available meta data

### PR DESCRIPTION
**Pull request by pstaeck**
The purpose of the pull request is to have a manipulation that returns the device meta data that must be available.

# Checklist

The following aspects have been respected by the author of this pull request, confirmed by both pull request assignee **and** reviewer:

* Changelog update (necessity checked and entry added or not added respectively)
  * [x] Pull Request Assignee
  * [x] Reviewer
* README update (necessity checked and entry added or not added respectively)
  * [x] Pull Request Assignee
  * [x] Reviewer

### CODE REVIEW

File: src/t2iapi/device/types.proto [hunk 1]
```diff
@@ -80,3 +82,26 @@ enum CalibrationState{
     CALIBRATION_STATE_OTHER = 4;
 }
 
+
+/*
+Contains the meta data of the device.
+ */
+message MetaDataMsg {
```
Comment (ldeichmann) [inline src/t2iapi/device/types.proto:89]: Why is this not in the biceps package, but all of the other types are? seems inconsistent to me

File: src/t2iapi/device/types.proto [hunk 2]
```diff
@@ -80,3 +82,26 @@ enum CalibrationState{
...
+    /*
+    Represents pm:Udi (defined in IEEE Std 11073-10207-2017).
+    */
+    message UdiMsg {
+      string device_identifier = 1;
+      string human_readable_form = 2;
+      t2iapi.biceps.InstanceIdentifierOneOfMsg issuer = 3;
+      t2iapi.biceps.InstanceIdentifierOneOfMsg jurisdiction = 4;
```
Comment (ldeichmann) [inline src/t2iapi/device/types.proto:97]: I think we can simplify here and only use InstanceIdentifier, not the related OneOf. The fact that OperatingJurisdiction is allowed is more like a bug in BICEPS and we should not make this more complicated for our users than it needs to be.

File: src/t2iapi/biceps/codedvalue.proto [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+Represents pm:CodedValue (defined in IEEE Std 11073-10207-2017).
+*/
+message CodedValueMsg {
+  repeated LocalizedTextMsg coding_system_name = 1;
+  repeated LocalizedTextMsg concept_description = 2;
+  /*
+  Represents pm:Translation (defined in IEEE Std 11073-10207-2017).
+  */
+  message TranslationMsg {
+    CodeIdentifierMsg code_attr = 3;
```
Comment (ldeichmann) [inline src/t2iapi/biceps/codedvalue.proto:30]: no need for this type indirection, it's a string

File: src/t2iapi/biceps/codedvalue.proto [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+    google.protobuf.StringValue coding_system_attr = 4;
+    google.protobuf.StringValue coding_system_version_attr = 5;
+  }
+  repeated TranslationMsg translation = 6;
+  CodeIdentifierMsg code_attr = 7;
```
Comment (ldeichmann) [inline src/t2iapi/biceps/codedvalue.proto:35]: no need for this type indirection, it's a string

File: src/t2iapi/biceps/codedvalue.proto [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+  google.protobuf.StringValue coding_system_attr = 8;
+  google.protobuf.StringValue coding_system_version_attr = 9;
+  SymbolicCodeNameMsg symbolic_code_name_attr = 10;
```
Comment (ldeichmann) [inline src/t2iapi/biceps/codedvalue.proto:38]: no need for this type indirection, it's a StringValue

File: src/t2iapi/biceps/localizedtext.proto [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+import "t2iapi/biceps/localizedtextref.proto";
+import "google/protobuf/wrappers.proto";
+import "t2iapi/biceps/referencedversion.proto";
+import "t2iapi/biceps/localizedtextwidth.proto";
+
+/*
+Represents pm:LocalizedText (defined in IEEE Std 11073-10207-2017).
+*/
+message LocalizedTextMsg {
+  LocalizedTextContentMsg localized_text_content = 1;
```
Comment (ldeichmann) [inline src/t2iapi/biceps/localizedtext.proto:25]: no need for this type indirection, it's a StringValue

File: src/t2iapi/biceps/localizedtext.proto [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+  LocalizedTextRefMsg ref_attr = 2;
```
Comment (ldeichmann) [inline src/t2iapi/biceps/localizedtext.proto:26]: no need for this type indirection, it's a StringValue

File: src/t2iapi/biceps/instanceidentifier.proto [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+message InstanceIdentifierMsg {
+  CodedValueMsg type = 1;
+  repeated LocalizedTextMsg identifier_name = 2;
+  /*
+  Represents pm:InstanceIdentifier/@Root (defined in IEEE Std 11073-10207-2017).
+  */
+  message RootAttrMsg {
+    string any_u_r_i = 3;
+  }
+  RootAttrMsg root_attr = 4;
```
Comment (ldeichmann) [inline src/t2iapi/biceps/instanceidentifier.proto:30]: no need for this type indirection, it's a StringValue

File: src/t2iapi/biceps/instanceidentifier.proto [hunk 9]
```diff
@@ -0,0 +1,39 @@
...
+  /*
+  Represents pm:InstanceIdentifier/@Extension (defined in IEEE Std 11073-10207-2017).
+  */
+  message ExtensionAttrMsg {
+    string string = 5;
+  }
+  ExtensionAttrMsg extension_attr = 6;
```
Comment (ldeichmann) [inline src/t2iapi/biceps/instanceidentifier.proto:37]: no need for this type indirection, it's a StringValue

File: src/t2iapi/biceps/localizedtext.proto [hunk 10]
```diff
@@ -0,0 +1,31 @@
...
+  google.protobuf.StringValue lang_attr = 3;
+  ReferencedVersionMsg version_attr = 4;
```
Comment (ldeichmann) [inline src/t2iapi/biceps/localizedtext.proto:28]: No need for this type indirection

File: src/t2iapi/device/types.proto (hunk 2 above)
Comment (pstaeck) [reply src/t2iapi/device/types.proto:97]: done

File: src/t2iapi/biceps/codedvalue.proto (hunk 3 above)
Comment (pstaeck) [reply src/t2iapi/biceps/codedvalue.proto:30]: done

File: src/t2iapi/biceps/codedvalue.proto (hunk 4 above)
Comment (pstaeck) [reply src/t2iapi/biceps/codedvalue.proto:35]: done

File: src/t2iapi/biceps/codedvalue.proto (hunk 5 above)
Comment (pstaeck) [reply src/t2iapi/biceps/codedvalue.proto:38]: done

File: src/t2iapi/biceps/localizedtext.proto (hunk 6 above)
Comment (pstaeck) [reply src/t2iapi/biceps/localizedtext.proto:25]: done

File: src/t2iapi/biceps/localizedtext.proto (hunk 7 above)
Comment (pstaeck) [reply src/t2iapi/biceps/localizedtext.proto:26]: done

Comment (pstaeck) [reply src/t2iapi/biceps/localizedtext.proto:26]: done

File: src/t2iapi/biceps/instanceidentifier.proto (hunk 9 above)
Comment (pstaeck) [reply src/t2iapi/biceps/instanceidentifier.proto:37]: done

File: src/t2iapi/biceps/localizedtext.proto (hunk 10 above)
Comment (pstaeck) [reply src/t2iapi/biceps/localizedtext.proto:28]: done - removed the indirection, and set it directly to uint64

File: src/t2iapi/biceps/instanceidentifier.proto (hunk 8 above)
Comment (pstaeck) [reply src/t2iapi/biceps/instanceidentifier.proto:30]: done

**Review by belagertem — APPROVED** [review]

File: src/t2iapi/biceps/codedvalue.proto (hunk 5 above)
Comment (ldeichmann) [reply src/t2iapi/biceps/codedvalue.proto]: Not really though, you made this into a string, not StringValue

File: src/t2iapi/biceps/localizedtext.proto (hunk 7 above)
Comment (ldeichmann) [reply src/t2iapi/biceps/localizedtext.proto]: Not really though, you made this into a string, not StringValue

File: src/t2iapi/biceps/instanceidentifier.proto (hunk 8 above)
Comment (ldeichmann) [reply src/t2iapi/biceps/instanceidentifier.proto]: Not really though, you made this into a string, not StringValue

File: src/t2iapi/biceps/instanceidentifier.proto (hunk 9 above)
Comment (ldeichmann) [reply src/t2iapi/biceps/instanceidentifier.proto]: Not really though, you made this into a string, not StringValue

File: src/t2iapi/biceps/localizedtext.proto (hunk 10 above)
Comment (ldeichmann) [reply src/t2iapi/biceps/localizedtext.proto]: Should be [UInt64Value](https://protobuf.dev/reference/protobuf/google.protobuf/#uint64-value) though.

**Review by ldeichmann — CHANGES_REQUESTED** [review]
Unresolved review remarks

File: src/t2iapi/biceps/codedvalue.proto (hunk 5 above)
Comment (pstaeck) [reply src/t2iapi/biceps/codedvalue.proto]: ah my fault... did not read correctly and changed it to String everywhere, just straight forward - done

File: src/t2iapi/biceps/localizedtext.proto (hunk 6 above)
Comment (pstaeck) [reply src/t2iapi/biceps/localizedtext.proto]: done

File: src/t2iapi/biceps/localizedtext.proto (hunk 7 above)
Comment (pstaeck) [reply src/t2iapi/biceps/localizedtext.proto]: done

File: src/t2iapi/biceps/instanceidentifier.proto (hunk 8 above)
Comment (pstaeck) [reply src/t2iapi/biceps/instanceidentifier.proto]: done

File: src/t2iapi/biceps/instanceidentifier.proto (hunk 9 above)
Comment (pstaeck) [reply src/t2iapi/biceps/instanceidentifier.proto]: done

File: src/t2iapi/biceps/localizedtext.proto (hunk 10 above)
Comment (pstaeck) [reply src/t2iapi/biceps/localizedtext.proto]: done

File: src/t2iapi/biceps/codedvalue.proto [hunk 11]
```diff
@@ -0,0 +1,37 @@
...
+option java_package = "com.draeger.medical.t2iapi.biceps";
+option java_outer_classname = "CodedValueProto";
+
+import "t2iapi/biceps/localizedtext.proto";
+import "google/protobuf/wrappers.proto";
+
+/*
+Represents pm:CodedValue (defined in IEEE Std 11073-10207-2017).
+*/
+message CodedValueMsg {
```
Comment (antipovi-draeger) [inline src/t2iapi/biceps/codedvalue.proto:21]: CodedValue is already implemeted, take a look if it can be used or maybe extended

Comment (pstaeck) [reply src/t2iapi/biceps/codedvalue.proto:21]: as discussed, we leave it as it is

File: src/t2iapi/biceps/localizedtextwidth.proto [hunk 12]
```diff
@@ -0,0 +1,33 @@
...
+*/
+
+syntax = "proto3";
+
+package t2iapi.biceps;
+
+option java_package = "com.draeger.medical.t2iapi.biceps";
+option java_outer_classname = "LocalizedTextWidthProto";
+
+
```
Comment (ldeichmann) [inline src/t2iapi/biceps/localizedtextwidth.proto:15]:
```suggestion

```

File: src/t2iapi/biceps/metadata.proto [hunk 13]
```diff
@@ -0,0 +1,43 @@
...
+option java_package = "com.draeger.medical.t2iapi.biceps";
+option java_outer_classname = "MetadataProto";
+
+import "t2iapi/biceps/localizedtext.proto";
+import "t2iapi/biceps/instanceidentifier.proto";
+import "google/protobuf/wrappers.proto";
+
+
+
+
```
Comment (ldeichmann) [inline src/t2iapi/biceps/metadata.proto:21]:
```suggestion

```

Comment (ldeichmann) [reply src/t2iapi/biceps/metadata.proto:21]: Weird diff, I just wanted to remove the blank lines

File: src/t2iapi/biceps/localizedtextwidth.proto (hunk 12 above)
Comment (pstaeck) [reply src/t2iapi/biceps/localizedtextwidth.proto:15]: done

File: src/t2iapi/biceps/metadata.proto (hunk 13 above)
Comment (pstaeck) [reply src/t2iapi/biceps/metadata.proto:21]: done

**Review by ldeichmann — APPROVED** [review]

**Pull request merged**